panda/src/navigation/navMesh.h

```diff
@@ -0,0 +1,69 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMesh.h
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+
+#ifndef NAVMESH_H
+#define NAVMESH_H
+
+#include "DetourNavMesh.h"
+#include "DetourNavMeshBuilder.h"
+#include "typedWritableReferenceCount.h"
+#include "pandaFramework.h"
+#include "pandaSystem.h"
+
+class EXPCL_NAVIGATION NavMesh: public TypedWritableReferenceCount
+{
+PUBLISHED:
+  NavMesh(dtNavMesh *nav_mesh);
+  void set_nav_mesh(dtNavMesh *m) { _nav_mesh = m; }
+  NavMesh();
+
+private:
+  dtNavMesh *_nav_mesh;
+  
+public:
+  bool init_nav_mesh();
+  dtNavMesh *get_nav_mesh() { return _nav_mesh; }
+  dtNavMeshCreateParams params;
```

Comment:
Member variables should have a leading underscore (i.e., `_params`). However, this may also be confusing with the "params" for BAM serialization. Maybe `_dt_params` or `_mesh_params` would be clearer? I would also prefer this to be private.

Reply:
Actually, I was calling this member and giving it values in NavMeshBuilder.cxx . So, I thought to keep it public.